### PR TITLE
Add PCL_UNUSED to fix doxygen warnings

### PR DESCRIPTION
**Pull request by aPonza**
Fixes `warning: argument <arg> of command @param is not found in the argument list of <function>`

If this seems reasonable I'll follow up with a PR aiming at using this macro instead of all the naked `(void)` casts.

### CODE REVIEW

File: common/include/pcl/pcl_macros.h [hunk 1]
```diff
@@ -123,6 +123,14 @@
...
                 major_version_mismatch)
 
+/**
+ * \brief Utility macro to eliminate unused variable warnings.
+ *
+ * \param x A parameter which will be used without being evaluated.
+ *
+ * Slightly better than `(void)x` as explained [here](https://stackoverflow.com/a/4030983/9926122).
+ */
+#define PCL_UNUSED(x) ((void)(true ? 0 : ((x), void(), 0)))
```
Comment (SergioRAgostinho) [inline common/include/pcl/pcl_macros.h:133]: My eyes are struggling to mentally compile this expression.

**Comment by SergioRAgostinho** [comment]
By the way this PR introduced warnings in 18.04
```
[ 42%] Building CXX object tools/CMakeFiles/pcl_plane_projection.dir/plane_projection.cpp.o                                                         
In file included from /home/sergio/Development/3rdparty/pcl/filters/include/pcl/filters/normal_refinement.h:40:0,          
                 from /home/sergio/Development/3rdparty/pcl/filters/include/pcl/filters/impl/normal_refinement.hpp:44,              
                 from /home/sergio/Development/3rdparty/pcl/filters/src/normal_refinement.cpp:39:                                   
/home/sergio/Development/3rdparty/pcl/filters/include/pcl/filters/normal_refinement.h: In instantiation of ‘std::vector<float> pcl::assignNormalWeights(const pcl::PointCloud<PointT>&, int, const
 std::vector<int>&, const std::vector<float>&) [with NormalT = pcl::Normal]’:                                                                                                                     
/home/sergio/Development/3rdparty/pcl/filters/include/pcl/filters/normal_refinement.h:102:60:   required from ‘bool pcl::refineNormal(const pcl::PointCloud<PointT>&, int, const std::vector<int>&
, const std::vector<float>&, NormalT&) [with NormalT = pcl::Normal]’                                                                                                                              
/home/sergio/Development/3rdparty/pcl/filters/include/pcl/filters/impl/normal_refinement.hpp:94:24:   required from ‘void pcl::NormalRefinement<NormalT>::applyFilter(pcl::NormalRefinement<Normal
T>::PointCloud&) [with NormalT = pcl::Normal; pcl::NormalRefinement<NormalT>::PointCloud = pcl::PointCloud<pcl::Normal>]’                                                                         
/home/sergio/Development/3rdparty/pcl/filters/src/normal_refinement.cpp:45:3:   required from here                                         
/home/sergio/Development/3rdparty/pcl/common/include/pcl/pcl_macros.h:133:46: warning: left operand of comma operator has no effect [-Wunused-value]
 #define PCL_UNUSED(x) ((void)(true ? 0 : ((x), void(), 0)))                                                                                        
                                           ~~~^~~~~~~~                                                   
/home/sergio/Development/3rdparty/pcl/filters/include/pcl/filters/normal_refinement.h:60:5: note: in expansion of macro ‘PCL_UNUSED’
     PCL_UNUSED(cloud);                                                                                                             
     ^                                                                                          
/home/sergio/Development/3rdparty/pcl/common/include/pcl/pcl_macros.h:133:46: warning: left operand of comma operator has no effect [-Wunused-value]
 #define PCL_UNUSED(x) ((void)(true ? 0 : ((x), void(), 0)))
                                           ~~~^~~~~~~~
/home/sergio/Development/3rdparty/pcl/filters/include/pcl/filters/normal_refinement.h:61:5: note: in expansion of macro ‘PCL_UNUSED’
     PCL_UNUSED(index);
```

**Comment by aPonza** [comment]
This actually didn't work for:
- GCC (both 5.5.0, and 9.2.1)
```
In file included from /__w/1/s/filters/include/pcl/filters/normal_refinement.h:40:0,
                 from /__w/1/s/filters/include/pcl/filters/impl/normal_refinement.hpp:44,
                 from /__w/1/s/filters/src/normal_refinement.cpp:39:
/__w/1/s/filters/include/pcl/filters/normal_refinement.h: In instantiation of 'std::vector<float> pcl::assignNormalWeights(const pcl::PointCloud<PointT>&, int, const std::vector<int>&, const std::vector<float>&) [with NormalT = pcl::Normal]':
/__w/1/s/filters/include/pcl/filters/normal_refinement.h:102:60:   required from 'bool pcl::refineNormal(const pcl::PointCloud<PointT>&, int, const std::vector<int>&, const std::vector<float>&, NormalT&) [with NormalT = pcl::Normal]'
/__w/1/s/filters/include/pcl/filters/impl/normal_refinement.hpp:94:24:   required from 'void pcl::NormalRefinement<NormalT>::applyFilter(pcl::NormalRefinement<NormalT>::PointCloud&) [with NormalT = pcl::Normal; pcl::NormalRefinement<NormalT>::PointCloud = pcl::PointCloud<pcl::Normal>]'
/__w/1/s/filters/src/normal_refinement.cpp:45:3:   required from here
/__w/1/s/common/include/pcl/pcl_macros.h:133:46: warning: left operand of comma operator has no effect [-Wunused-value]
 #define PCL_UNUSED(x) ((void)(true ? 0 : ((x), void(), 0)))
                                              ^
/__w/1/s/filters/include/pcl/filters/normal_refinement.h:60:5: note: in expansion of macro 'PCL_UNUSED'
     PCL_UNUSED(cloud);
     ^
/__w/1/s/common/include/pcl/pcl_macros.h:133:23: warning: left operand of comma operator has no effect [-Wunused-value]
 #define PCL_UNUSED(x) ((void)(true ? 0 : ((x), void(), 0)))
                       ^
/__w/1/s/filters/include/pcl/filters/normal_refinement.h:61:5: note: in expansion of macro 'PCL_UNUSED'
     PCL_UNUSED(index);
     ^
```
([link](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=15780&view=logs&j=d5940dd1-f907-59fe-4c1f-39558d633862&t=a45f8aa0-5d63-5c3a-dbc2-c6356a830645&l=2315))

- XCode
```
In file included from /Users/runner/runners/2.166.2/work/1/s/filters/src/normal_refinement.cpp:39:
In file included from /Users/runner/runners/2.166.2/work/1/s/filters/include/pcl/filters/impl/normal_refinement.hpp:44:
/Users/runner/runners/2.166.2/work/1/s/filters/include/pcl/filters/normal_refinement.h:61:16: warning: expression result unused [-Wunused-value]
    PCL_UNUSED(index);
               ^~~~~
/Users/runner/runners/2.166.2/work/1/s/common/include/pcl/pcl_macros.h:133:44: note: expanded from macro 'PCL_UNUSED'
#define PCL_UNUSED(x) ((void)(true ? 0 : ((x), void(), 0)))
                                           ^
```
([link](https://dev.azure.com/PointCloudLibrary/pcl/_build/results?buildId=15780&view=logs&j=eb6fcca7-7d1a-5fb6-a052-b530c1724910&t=9653028d-5065-59e7-e30e-ad178b71bee8&l=942))

But it was perfect for Doxygen and MSVC. Maybe for the rest `(void)x` will do since it's in place everywhere else in the library. This really should've been a draft PR.

**Comment by aPonza** [comment]
Didn't update the page so I didn't see your message.

Anyways [coliru](http://coliru.stacked-crooked.com/a/ac86ff6074b30aea) seems ok with clang and gcc, and [godbolt](https://godbolt.org/z/kakpXY) likes the MSVC version. I don't have anything online to test XCode, so I'll check that Doxygen doesn't complain and push the fix.

**Pull request merged**